tools/ci_setup.ml

```diff
@@ -0,0 +1,296 @@
+
+let roots = [ "bonsai"; "string_dict" ]
+
+let omitted_others = StringSet.of_list [ "cohttp-async"; "cohttp"; "uri"; "uri-sexp" ]
+
+let omitted_js = StringSet.of_list [ "sexplib0" ]
+
+let do_not_pin = StringSet.of_list [ "wasocaml"; "wasm_of_ocaml"; "dune" ]
+
+let do_pin = StringSet.of_list [ "base"; "ppx_expect"; "ppx_inline_test"; "time_now" ]
+
+let aliases = [ "ocaml-cstruct", "cstruct" ]
+
+let dune_workspace =
+  {|(lang dune 3.17)
+(env
+ (_
+  (env-vars (TESTING_FRAMEWORK inline-test))
+  (js_of_ocaml (enabled_if false))
+  (flags :standard -warn-error -8-32-34-49-52-55 -w -67-69)))
+|}
+
+let patches =
+  [ ( "sexp_grammar"
+    , {|
+diff --git a/sexp_grammar_validation.opam b/sexp_grammar_validation.opam
+new file mode 100644
+index 0000000..e69de29
+diff --git a/validation/src/dune b/validation/src/dune
+index 91933ec..849e4d7 100644
+--- a/validation/src/dune
++++ b/validation/src/dune
+@@ -1,5 +1,6 @@
+ (library
+  (name sexp_grammar_validation)
++ (public_name sexp_grammar_validation)
+  (libraries bignum.bigint core
+    expect_test_helpers_core.expect_test_helpers_base sexp_grammar)
+  (preprocess
+|}
+    )
+  ; ( "bignum"
+    , {bignum|
+diff --git a/dune-project b/dune-project
+index e563d7e..b87e356 100644
+--- a/dune-project
++++ b/dune-project
+@@ -1,3 +1,3 @@
+-(lang dune 2.0)
++(lang dune 3.17)
+ 
+ (formatting disabled)
+diff --git a/test/src/dune b/test/src/dune
+index f93ae3f..3f00557 100644
+--- a/test/src/dune
++++ b/test/src/dune
+@@ -2,5 +2,6 @@
+  (name bignum_test)
+  (libraries bigint bignum core expect_test_helpers_core
+    sexp_grammar_validation)
++ (inline_tests (flags -drop-tag no-js -drop-tag no-wasm -drop-tag 64-bits-only) (modes js wasm))
+  (preprocess
+   (pps ppx_jane)))
+diff --git a/test/src/test_bignum.ml b/test/src/test_bignum.ml
+index c6d09fb..61b1e5b 100644
+--- a/test/src/test_bignum.ml
++++ b/test/src/test_bignum.ml
+@@ -3,6 +3,11 @@ open! Expect_test_helpers_core
+ open Bignum
+ open Bignum.For_testing
+ 
++module Zarith = struct
++  module Q = Q
++  module Z = Z
++end
++
+ let%expect_test "Bignum.abs" =
+   let test t =
+     let t' = require_no_allocation [%here] (fun () -> abs t) in
+@@ -71,7 +76,7 @@ let%expect_test "Bignum.sexp_of_t does use Scientific Notation" =
+ let compare_floats ~of_float x =
+   let x' = x |> of_float |> Bignum.to_float in
+   if not (Float.( = ) x x' || (Float.is_nan x && Float.is_nan x'))
+-  then raise_s [%message "mismatch" (x : float) (x' : float)]
++  then raise_s [%message "mismatch" (x : float) (x' : float) (x |> of_float : Bignum.t)]
+ ;;
+ 
+ let%expect_test "roundtrip: f |> Bignum.of_float_decimal |> Bignum.to_float" =
+@@ -783,7 +788,7 @@ let%test_module _ =
+          -1073741825 -> ( 6) \001\253\255\255\255\191 |}]
+     ;;
+ 
+-    let%expect_test ("bin_io serialization V2 (javascript)" [@tags "js-only"]) =
++    let%expect_test ("bin_io serialization V2 (javascript)" [@tags "js-only", "no-wasm"]) =
```

Comment:
Why is the expect test different in js and wasm ?

Reply:
The integer size differs.